File: GXpert/GXpert.Web/Modules/Masters/UserAcademic/UserAcademicForm.cs
using Serenity.ComponentModel;

namespace GXpert.Masters.Forms;

[FormScript("Masters.UserAcademic")]
[BasedOnRow(typeof(UserAcademicRow), CheckNames = true)]
public class UserAcademicForm
{
    public int StudentId { get; set; }
    public int AcademicYearId { get; set; }
    public string Title { get; set; }
    [TextAreaEditor(Rows =3)]
    public string Description { get; set; }
    public bool IsVerify { get; set; }
    public int VerifyBy { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/UserAcademicForm.ts
import { LookupEditor, StringEditor, TextAreaEditor, BooleanEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface UserAcademicForm {
    StudentId: LookupEditor;
    AcademicYearId: LookupEditor;
    Title: StringEditor;
    Description: TextAreaEditor;
    IsVerify: BooleanEditor;
    VerifyBy: IntegerEditor;
}

export class UserAcademicForm extends PrefixedContext {
    static readonly formKey = 'Masters.UserAcademic';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UserAcademicForm.init)  {
            UserAcademicForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;
            var w3 = BooleanEditor;
            var w4 = IntegerEditor;

            initFormType(UserAcademicForm, [
                'StudentId', w0,
                'AcademicYearId', w0,
                'Title', w1,
                'Description', w2,
                'IsVerify', w3,
                'VerifyBy', w4
            ]);
        }
    }
}